src/app/shared/focus-element/focus-element.ts

```diff
@@ -0,0 +1,26 @@
+import {Renderer, Directive, ElementRef} from '@angular/core';
+
+
+@Directive({
+  selector: '[focusElement]'
+})
+
+export class FocusElementDirective {
+
+  constructor(
+    private _element: ElementRef,
+    private _renderer: Renderer
+  ) { }
+
+  ngAfterViewInit() {
+    // Add tabindex 0 so element is focusable by keyboard
+    this
+      ._renderer
+      .setElementAttribute(this._element.nativeElement, 'tabindex', '0');
+
+    // Focus host element after view loads
+    // so screen readers will be alerted
+    // the page has changed.
+    this._element.nativeElement.focus();
```

Comment:
@riavalon have you tried setting a document title? I would expect a screen reader to notice and announce a change in the title. Have a look here: https://angular.io/docs/ts/latest/cookbook/set-document-title.html
Can you see if this is a better solution?